tls/s2n_x509_validator.c

```diff
@@ -549,3 +554,66 @@ s2n_cert_validation_code s2n_x509_validator_validate_cert_stapled_ocsp_response(
     return ret_val;
 #endif /* S2N_OCSP_STAPLING_SUPPORTED */
 }
+
+S2N_RESULT s2n_validate_certificate_signature(struct s2n_connection *conn, X509 *x509_cert)
+{
+    ENSURE_REF(conn);
+    ENSURE_REF(x509_cert);
+
+    const struct s2n_security_policy *security_policy;
+    GUARD_AS_RESULT(s2n_connection_get_security_policy(conn, &security_policy));
+
+    if (security_policy->certificate_signature_preferences == NULL) {
+        return S2N_RESULT_OK;
+    }
```

Comment:
Do you have a test for this case?

Reply:
The test for this case is here: https://github.com/awslabs/s2n/pull/2370/files#diff-793694a75b3296c1444255e001b219018ac01928f80fc0d7a0919a933c5396ffR170

Reply:
Then I'd update the comment/name of that test to make it clear we're interesting in a security policy with NULL certificate signature preferences. Otherwise, if we ever add certificate signature preferences to "default", we'd probably update the expected outcome of that test rather than switching it to testing a different security policy :)

It's also probably a good idea just to not use "default", since it can change. Pin the test to a specific security policy you know has no certificate signature preferences.